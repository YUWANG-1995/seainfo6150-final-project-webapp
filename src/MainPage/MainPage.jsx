import React from "react";
import styles from "./MainPage.module.css";
import { Link } from "react-router-dom";
import bmw from "../images/MainPage/bmw_i8.png";
import audi from "../images/MainPage/Audi.jpg";

const MainPage = () => {
  return (
    <div>
      <div className={styles.descripePart}>
        <h1>Lexi's Car Shop </h1>
        <h4>Meet You Match</h4>
        <p className={styles.pDetail}>While every car in the Lexi'S family offers innovative technology, exceptional efficiency and impressive safety features, each one has a personality all its own. Whether you’re looking for a spunky sidekick for your next urban adventure or a sophisticated sedan for nights out on the town, there’s a Chevrolet car that’s a perfect fit.</p>
      </div>
      <div className={styles.gridContainer}>
        <div>
          <Link to="/cars/BMW">
            <img className={styles.image} src={bmw} alt="bmw i8 "/>
          </Link>
          <div className={styles.gridFont}>
            <p>BMW i8 </p>
            <p> $100,000+ </p>
            <p>Lowest Price</p>
            <p>up to 80% off!</p>
          </div>
        </div>
        <div>
          <Link to="/cars/Audi">
            <img className={styles.image}  src={audi} alt="audi r8 "/>
          </Link>
          <div className={styles.gridFont}>
            <p>Audi R8</p>
            <p>$100,000+</p>
            <p>Lowest Price</p>
            <p>up to 85% off!</p>
          </div>
        </div>
      </div>
    </div>
  )
}

export default MainPage;
