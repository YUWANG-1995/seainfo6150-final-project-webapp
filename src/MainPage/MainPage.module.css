.descripePart{
  display: flex;
  flex-direction: column;
  align-items: center;

  font-family: 'Roboto', sans-serif;
}

.pDetail{
  font-style: italic;
  letter-spacing: 1px;
}

.gridContainer{
  display: grid;
  grid-template-columns: 1fr 1fr;
  column-gap: 10px;
}

.image{
  width: 600px;
  height: 400px;
}


.gridFont{
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: 1fr 1fr;
}
