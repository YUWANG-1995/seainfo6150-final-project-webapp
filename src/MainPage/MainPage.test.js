import React from "react";
import { render } from "@testing-library/react";
import MainPage from "./MainPage.jsx";
import { BrowserRouter } from 'react-router-dom'

it("renders the MainPage Component Correctly", () => {

  const { container } = render(
    <BrowserRouter>
      <MainPage />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
