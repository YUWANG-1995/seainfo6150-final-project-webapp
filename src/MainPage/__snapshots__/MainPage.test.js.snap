// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the MainPage Component Correctly 1`] = `
<div>
  <div>
    <div
      class="descripePart"
    >
      <h4>
        Meet You Match
      </h4>
      <p
        class="pDetail"
      >
        While every car in the Lexi'S family offers innovative technology, exceptional efficiency and impressive safety features, each one has a personality all its own. Whether you’re looking for a spunky sidekick for your next urban adventure or a sophisticated sedan for nights out on the town, there’s a Chevrolet car that’s a perfect fit.
      </p>
    </div>
    <div
      class="gridContainer"
    >
      <div>
        <a
          href="/cars/BMW"
        >
          <img
            alt="bmw i8 "
            class="image"
            src="bmw_i8.png"
          />
        </a>
        <div
          class="gridFont"
        >
          <p>
            BMW i8 
          </p>
          <p>
             $100,000+ 
          </p>
          <p>
            Lowest Price
          </p>
          <p>
            up to 80% off!
          </p>
        </div>
      </div>
      <div>
        <a
          href="/cars/Audi"
        >
          <img
            alt="audi r8 "
            class="image"
            src="Audi.jpg"
          />
        </a>
        <div
          class="gridFont"
        >
          <p>
            Audi R8
          </p>
          <p>
            $100,000+
          </p>
          <p>
            Lowest Price
          </p>
          <p>
            up to 85% off!
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
